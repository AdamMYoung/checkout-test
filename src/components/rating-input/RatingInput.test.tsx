import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { axe } from 'jest-axe';
import { RatingInput } from '.';

describe('RatingInput', () => {
    it('passes aXe validation', async () => {
        const { container } = render(<RatingInput />);

        expect(await axe(container)).toHaveNoViolations();
    });

    it('matches snapshot', () => {
        const { container } = render(<RatingInput />);

        expect(container).toMatchSnapshot();
    });

    it('fires the correct event when a star is selected', () => {
        const onChange = jest.fn();
        render(<RatingInput onChange={onChange} />);

        const ratingElement = screen.getByLabelText(/4/i);
        userEvent.click(ratingElement);

        expect(onChange).toBeCalledTimes(1);
        expect(onChange).toBeCalledWith('4');
        expect(ratingElement).toBeInTheDocument();
    });
});
