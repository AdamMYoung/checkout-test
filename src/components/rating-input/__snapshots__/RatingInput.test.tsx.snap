// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingInput matches snapshot 1`] = `
<div>
  <div
    class="css-nn803e"
    role="radiogroup"
  >
    <label
      class="css-0"
    >
      <input
        aria-label="1 star"
        id="radio-7"
        name="radio-12"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="1"
      />
      <svg
        aria-hidden="true"
        class="chakra-icon css-16r67my"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
          fill="currentColor"
        />
      </svg>
    </label>
    <label
      class="css-0"
    >
      <input
        aria-label="2 star"
        id="radio-8"
        name="radio-12"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="2"
      />
      <svg
        aria-hidden="true"
        class="chakra-icon css-16r67my"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
          fill="currentColor"
        />
      </svg>
    </label>
    <label
      class="css-0"
    >
      <input
        aria-label="3 star"
        id="radio-9"
        name="radio-12"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="3"
      />
      <svg
        aria-hidden="true"
        class="chakra-icon css-16r67my"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
          fill="currentColor"
        />
      </svg>
    </label>
    <label
      class="css-0"
    >
      <input
        aria-label="4 star"
        id="radio-10"
        name="radio-12"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="4"
      />
      <svg
        aria-hidden="true"
        class="chakra-icon css-16r67my"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
          fill="currentColor"
        />
      </svg>
    </label>
    <label
      class="css-0"
    >
      <input
        aria-label="5 star"
        id="radio-11"
        name="radio-12"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="5"
      />
      <svg
        aria-hidden="true"
        class="chakra-icon css-16r67my"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
          fill="currentColor"
        />
      </svg>
    </label>
  </div>
</div>
`;
